modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryReaderHandlesHolder.java

```diff
@@ -21,6 +21,15 @@
  * Holder for handles.
  */
 interface BinaryReaderHandlesHolder {
+    /**
+     * Set handle.
+     *
+     * @param obj Object.
+     */
+    public default void setHandle(Object obj) {
```

Comment:
Default implementation used only in single class - BinaryReaderHandlesHolderImpl. Let's move this exception to this class